### ISSUE TITLE
Collection::implode() with last item separator support

### ISSUE THREAD

**Issue by rentalhost** [laravel/ideas#1109]
Add a third parameter to `Collection::implode()` to works like a last item separator ("and", "or"). It should be like `Collection::implode($value, $glue = null, $lastGlue = null)`. 

## Behavior 

1. Will works only if  `$lastGlue` is not `null`, else will works exactly as currently;
2. Will works only if `count($this->items) >= 2`;
3. Will get `$this->last()` and will implode all items except last;
4. Will concatenate previous value with the `$lastGlue` and last item;

## Example

```php
$collection = new Collection([ '1', '2' ]);

$collection->implode(', ');          // string ("1, 2")
$collection->implode(', ', ' and '); // string ("1 and 2")

$collection->push('3');

$collection->implode(', ');          // string ("1, 2, 3")
$collection->implode(', ', ' and '); // string ("1, 2 and 3")
```

## Example Macro

Currently I am working with macros, but I think that it could be done by the framework once that is an easy and very useful feature. That is my current version (it have some IDE annotations).

```php
Collection::macro('implodeLast', function ($value, $glue = null, $lastGlue = null) {
    /** @var Collection $collection */
    $collection = $this;
    $first      = $collection->first();

    if (\is_array($first) || \is_object($first)) {
        return $collection->pluck($value)->implodeLast($glue, $lastGlue);
    }

    if ($collection->count() >= 2) {
        return $collection->slice(0, -1)->implode($value) . $glue . $collection->last();
    }

    return implode($value, $collection->all());
});
```

References:

* https://stackoverflow.com/questions/8586141/implode-array-with-and-add-and-before-last-item
* https://stackoverflow.com/questions/18476587/comma-separated-list-from-array-with-and-before-last-element
* https://stackoverflow.com/questions/7687665/replace-the-last-comma-with-an-sign

**Comment by staudenmeir** [comment]
https://github.com/laravel/framework/pull/27723 added the `join()` method with a `$finalGlue` parameter.

**Comment by rentalhost** [comment]
You are right.

Closing. Implemented and merged by #27723.
